src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -21,21 +21,25 @@ namespace Microsoft.Azure.Templates.Analyzer.Core
     /// </summary>
     public class TemplateAnalyzer
     {
-        private JsonRuleEngine jsonRuleEngine { get; }
+        private JsonRuleEngine jsonRuleEngine;
+        private PowerShellRuleEngine powerShellRuleEngine;
 
         /// <summary>
         /// Private constructor to enforce use of <see cref="TemplateAnalyzer.Create"/> for creating new instances.
         /// </summary>
         /// <param name="jsonRuleEngine">The <see cref="JsonRuleEngine"/> to use in analyzing templates.</param>
-        private TemplateAnalyzer(JsonRuleEngine jsonRuleEngine)
+        /// <param name="powerShellRuleEngine">The <see cref="PowerShellRuleEngine"/> to use in analyzing templates.</param>
+        private TemplateAnalyzer(JsonRuleEngine jsonRuleEngine, PowerShellRuleEngine powerShellRuleEngine)
         {
             this.jsonRuleEngine = jsonRuleEngine;
+            this.powerShellRuleEngine = powerShellRuleEngine;
         }
 
         /// <summary>
         /// Creates a new <see cref="TemplateAnalyzer"/> instance with the default built-in rules.
         /// </summary>
-        public static TemplateAnalyzer Create()
+        /// <param name="usePowerShell">Whether or not to use PowerShell rules to analyze the template.</param>
+        public static TemplateAnalyzer Create(bool usePowerShell)
```

Comment:
Is it worth putting a default value? That way in the tests you don't have to put "false"

Reply:
I think I'll leave this without a default parameter.